### PR TITLE
Update TargetFinder

### PR DESCRIPTION
**Pull request by gallardoalba**
This PR updates the TargetFinder tool.
- Replace the previous output format (data) with the correct txt extension.
- Incorporate the option of generating the outputs in different formats (gff, json, tabular)
- Update the wrapper style.
- Include additional tests.
- Improve the help section.

### CODE REVIEW

File: tools/rna_tools/targetfinder/macros.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            <edam_topic>topic_3170</edam_topic>
+            <edam_topic>topic_3308</edam_topic>
+        </edam_topics>
+        <edam_operations>
+            <edam_operation>operation_0292</edam_operation>
+        </edam_operations>
+    </xml>
+    <xml name='requirements'>
+        <requirements>
+            <requirement type='package' version='1.7'>targetfinder</requirement>
```
Comment (bgruening) [inline tools/rna_tools/targetfinder/macros.xml:14]: TOOL_VERSION can be reused here.

File: tools/rna_tools/targetfinder/TargetFinder.xml [hunk 2]
```diff
@@ -1,73 +1,236 @@
-<tool id="targetfinder" name="Plant small RNA target prediction tool" version="1.7.0">
```
Comment (bgruening) [inline tools/rna_tools/targetfinder/TargetFinder.xml:1]: Cris please note that the old version was 1.7.0 the new one will be 1.7 ... this is probably lower? Not sure. Maybe we need to hardcode the version here to 1.7.0+galaxy1 or something like that to make sure its really bigger than the previous one.

File: tools/rna_tools/targetfinder/TargetFinder.xml [hunk 3]
```diff
@@ -1,73 +1,236 @@
...
+        <param name='output_format' argument='-p' type='select' label='Output format' help='Output format for small RNA-target pairs (Default = classic).' >
+                <option value='classic'>Original TargetFinder base-pair format</option>
+                <option value='gff'>GFF</option>
+                <option value='json'>JavaScript Object Notation (JSON)</option>
+                <option value='table'>Tab-delimited format</option>
+        </param>
     </inputs>
     <outputs>
-        <data name="output1" format="data"/>
+        <data name='output_tabular' format='tabular'>
```
Comment (bgruening) [inline tools/rna_tools/targetfinder/TargetFinder.xml:43]: You can also do it this way: https://github.com/bgruening/galaxytools/blob/dee6a6f4aa8cbee9f81692fc6f3a2e4ad1f3abf6/tools/rest_tool/rest_tool.xml#L192

File: tools/rna_tools/targetfinder/TargetFinder.xml [hunk 4]
```diff
@@ -1,73 +1,236 @@
...
+    </macros>
+    <expand macro='edam' />
+    <expand macro='requirements' />
     <stdio>
-        <exit_code range="1:" />
+        <exit_code range='1:' />
     </stdio>
     <command><![CDATA[
         targetfinder_threads.pl
             -f $f
```
Comment (bgruening) [inline tools/rna_tools/targetfinder/TargetFinder.xml:13]: f and d needs to be quoted

File: tools/rna_tools/targetfinder/TargetFinder.xml [hunk 5]
```diff
@@ -1,73 +1,236 @@
...
             -d $d
             -c $c
+            -p $output_format
             -t "\${GALAXY_SLOTS:-12}"
             $r
-            -o $output1
+            #if $output_format == 'classic'
+                -o $output_text
```
Comment (bgruening) [inline tools/rna_tools/targetfinder/TargetFinder.xml:20]: all single-quoted

**Review by bgruening — COMMENTED** [review]
Cool, nice!

File: tools/rna_tools/targetfinder/TargetFinder.xml (hunk 3 above)
Comment (gallardoalba) [reply tools/rna_tools/targetfinder/TargetFinder.xml:43]: Great, nice.

**Comment by bgruening** [comment]
Nice thanks, will be installed Saturday.

**Pull request merged**